From cce10cf4dc594a4ab7c17508680074f945ababdc Mon Sep 17 00:00:00 2001
From: topjohnwu <topjohnwu@gmail.com>
Date: Sun, 14 Jul 2024 16:55:01 -0700
Subject: [PATCH 06/24] Add missing sigisemptyset implementation

---
 shell/hush.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/shell/hush.c b/shell/hush.c
index 707b77c..fe6c1db 100644
--- a/shell/hush.c
+++ b/shell/hush.c
@@ -1961,6 +1961,15 @@ enum {
 		,
 };
 
+#ifdef __BIONIC__
+static int sigisemptyset(const sigset_t *set)
+{
+    sigset_t empty;
+    sigemptyset(&empty);
+    return memcmp(&empty, set, sizeof(sigset_t)) == 0;
+}
+#endif
+
 static void record_pending_signo(int sig)
 {
 	sigaddset(&G.pending_set, sig);
-- 
2.45.2

