From 4530fad34caf60516e0fcf2ad5f0f2ded9251b60 Mon Sep 17 00:00:00 2001
From: Chris Renshaw <osm0sis@outlook.com>
Date: Wed, 24 May 2023 14:15:00 -0300
Subject: [PATCH 10/24] Fix direct mount auto loopback and losetup for Android

---
 include/libbb.h | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/include/libbb.h b/include/libbb.h
index 01cdb1b..8b1d49f 100644
--- a/include/libbb.h
+++ b/include/libbb.h
@@ -2414,9 +2414,15 @@ extern const char bb_default_login_shell[] ALIGN1;
 # define VC_4 "/dev/tty4"
 # define VC_5 "/dev/tty5"
 # define VC_FORMAT "/dev/tty%d"
-# define LOOP_FORMAT "/dev/loop%u"
-# define LOOP_NAMESIZE (sizeof("/dev/loop") + sizeof(int)*3 + 1)
-# define LOOP_NAME "/dev/loop"
+# if defined(__ANDROID__)
+#  define LOOP_FORMAT "/dev/block/loop%u"
+#  define LOOP_NAMESIZE (sizeof("/dev/block/loop") + sizeof(int)*3 + 1)
+#  define LOOP_NAME "/dev/block/loop"
+# else
+#  define LOOP_FORMAT "/dev/loop%u"
+#  define LOOP_NAMESIZE (sizeof("/dev/loop") + sizeof(int)*3 + 1)
+#  define LOOP_NAME "/dev/loop"
+# endif
 # define FB_0 "/dev/fb0"
 #endif
 
-- 
2.45.2

